src/reducers/columnStateHelper-test.js

```diff
@@ -0,0 +1,219 @@
+            }
+          }]
+        },
+        isColumnReordering: true,
+        columnReorderingData: {
+          dragDistance: 0,
+          isFixed: true,
+          scrollStart: 50,
+          columnKey: 'col1',
+          columnWidth: 100,
+          originalLeft: 20,
+          columnBefore: undefined,
+          columnAfter: undefined
+        }
+      });
+    });
+  });
+
+  describe('reorderColumnMove', function() {
+    it('should update column reorder data based on new position', function() {
+      var oldState = {
+        columnInfo: {
+          bodyFixedColumns: [{
+            props: {
+              width: 250
+            }
+          }]
+        },
+        scrollX: 30,
+        maxScrollX: 50,
+        width: 550,
+        isColumnReordering: true,
+        columnReorderingData: {
+          dragDistance: 0,
+          isFixed: false,
+          scrollStart: 50,
+          columnKey: 'col1',
+          columnWidth: 100,
+          originalLeft: 20,
+          columnBefore: undefined,
+          columnAfter: undefined
+        }
+      };
+
+      // --- Run Test ---
+      var result = columnStateHelper.reorderColumnMove(oldState, 277);
+
+      // --- Verify Expectations ---
+      assert.deepEqual(result, {
+        columnInfo: {
+          bodyFixedColumns: [{
+            props: {
+              width: 250
+            }
+          }]
+        },
+        scrollX: 45,
+        maxScrollX: 50,
+        width: 550,
+        isColumnReordering: true,
+        columnReorderingData: {
+          dragDistance: 257,
+          isFixed: false,
+          scrollStart: 50,
+          columnKey: 'col1',
+          columnWidth: 100,
+          originalLeft: 20,
+          columnBefore: undefined,
+          columnAfter: undefined
+        }
+      });
+    });
+  });
+
+  describe('initialize', function() {
+    it('should initialize column state as expected', function() {
+      var oldState = {
+        scrollX: 30,
+        isColumnReordering: true,
+        columnReorderingData: {
+          placeholder: true
+        },
+        isColumnResizing: true,
+        columnResizingData: {
+          placeholder: true
+        }
+      };
+
+      var children = [
+        React.createElement(FixedDataTableColumn, {
+          columnKey: 'fixed1',
+          fixed: true,
+          width: 150
+        }),
+        React.createElement(FixedDataTableColumn, {
+          columnKey: 'fixed2',
+          fixed: true,
+          width: 75
+        }),
+        React.createElement(FixedDataTableColumn, {
+          columnKey: 'scrollable1',
+          fixed: false,
+          width: 12
+        }),
+        React.createElement(FixedDataTableColumn, {
+          columnKey: 'scrollable2',
+          fixed: false,
+          width: 200
+        })
+      ];
+
+      // --- Run Test ---
+      var result = columnStateHelper.initialize(oldState, {
+        children: children,
+        width: 200,
+        isColumnResizing: true
```

Comment:
This seems analogous to the other props, isColumnReordering and isColumnResizing. I'm not sure why they are there but we shouldn't break the API yet